pettingzoo/classic/rlcard_envs/dou_dizhu.py

```diff
@@ -34,12 +34,13 @@ def _scale_rewards(self, reward):
     def observe(self, agent):
         obs = self.env.get_state(self._name_to_int(agent))
         if self._opponents_hand_visible:
-            observation = obs['obs'].astype(self._dtype)
+            observation = np.zeros(901, dtype=self._dtype)
+            observation[:len(obs['obs'])] = obs['obs'].astype(self._dtype)
         else:
-            observation = obs['obs'][[0, 2, 3, 4], :, :].astype(self._dtype)
+            observation = np.concatenate((obs['obs'][:54].astype(self._dtype), obs['obs'][108:648].astype(self._dtype)))
```

Comment:
What observation features are being extracted here? I think that the new version of dou dizhu has different observation sizes for the landlord and for the peasants.

Reply:
So here I'm extracting `current_hand`, `last_action` and `last_9_actions` whether it's a landlord or peasant. They have the same indices for those states.